addons/resource_distribution_controller.go

```diff
@@ -0,0 +1,161 @@
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("unable to distribute resources to StorageConsumers: %w", err)
+	}
+	if len(storageConsumerList.Items) == 0 {
+		logger.Info("No StorageConsumer to add/remove resources.")
+		if slices.Contains(addonDeletionlock.GetFinalizers(), ResourceDistributionFinalizer) {
+			logger.Info("Removing finalizer from addonDeletionlock", "finalizer", ResourceDistributionFinalizer)
+			addonDeletionlock.Finalizers = utils.RemoveString(addonDeletionlock.Finalizers, ResourceDistributionFinalizer)
+			if err := r.SpokeClient.Update(ctx, &addonDeletionlock); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+		return ctrl.Result{}, nil
+	}
+
+	err = r.SpokeClient.List(ctx, &templateList, client.InNamespace(r.CurrentNamespace), client.MatchingLabels{
+		utils.CreatedByLabelKey: "odf-multicluster-orchestrator",
+	})
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("unable to distribute resources to StorageConsumers: %w", err)
+	}
+	if len(templateList.Items) == 0 {
+		for _, consumer := range storageConsumerList.Items {
+			consumer.Spec.VolumeReplicationClasses = []ocsv1alpha1.VolumeReplicationClassSpec{}
+			err := r.SpokeClient.Update(ctx, &consumer)
+			if err != nil {
+				return ctrl.Result{}, fmt.Errorf("unable to remove VolumeReplicationClasses from StorageConsumer %q: %w", consumer.Name, err)
+			}
+			logger.Info("All VolumeReplicationClasses were removed from StorageConsumer.", "StorageConsumer", consumer.Name)
+		}
+		if slices.Contains(addonDeletionlock.GetFinalizers(), ResourceDistributionFinalizer) {
+			logger.Info("Removing finalizer from addonDeletionlock", "finalizer", ResourceDistributionFinalizer)
+			addonDeletionlock.Finalizers = utils.RemoveString(addonDeletionlock.Finalizers, ResourceDistributionFinalizer)
+			if err := r.SpokeClient.Update(ctx, &addonDeletionlock); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+		return ctrl.Result{}, nil
+	}
+
+	var storageClientMapping *corev1.ConfigMap
+	storageClientMapping, err = utils.GetStorageClientMapping(ctx, r.SpokeClient, r.CurrentNamespace)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("unable to distribute resources to StorageConsumers: %w", err)
+	}
+
+	for k := range storageClientMapping.Data {
+		for consumerIndex := range storageConsumerList.Items {
+			if storageConsumerList.Items[consumerIndex].Status.Client.ID == k {
+				addVRCToConsumers = append(addVRCToConsumers, &storageConsumerList.Items[consumerIndex])
+			}
+		}
+	}
+
+	for _, foundConsumer := range addVRCToConsumers {
+		expectedConsumer := foundConsumer.DeepCopy()
+		expectedConsumer.Spec.VolumeReplicationClasses = []ocsv1alpha1.VolumeReplicationClassSpec{}
+		for i := range templateList.Items {
+			vrc := ocsv1alpha1.VolumeReplicationClassSpec{Name: templateList.Items[i].Name}
+			expectedConsumer.Spec.VolumeReplicationClasses = append(expectedConsumer.Spec.VolumeReplicationClasses, vrc)
```

Comment:
This will append all the VRC that is present in the cluster to the list, which means that if a client doesn't require a VRC they will still get it

Reply:
This is expected. Clients having VRCs even if they don't need it is a non-issue. Doing so simplifies logic on our end. It is important to note that, only clients that have DR enabled will get the VRC. So it's not so bad overall.